scipy/optimize/_simulated_annealing.py

```diff
@@ -0,0 +1,559 @@
+# Simulated Dual Annealing implementation.
+# Copyright (c) 2018 Sylvain Gubian <sylvain.gubian@pmi.com>,
+# Yang Xiang <yang.xiang@pmi.com>
+# Author: Sylvain Gubian, PMP S.A.
+"""
+SDA: A Simulated Dual Annealing global optimization algorithm
+"""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.optimize import OptimizeResult
+from scipy.optimize import minimize
+from scipy.special import gammaln
+from scipy._lib._util import check_random_state
+
+__all__ = ['simulated_annealing']
+BIG_VALUE = 1e16
+
+
+class VisitingDistribution(object):
+    """
+    Class used to generate new coordinates based on the distorted
+    Cauchy-Lorentz distribution. Depending on the steps within the Markov
+    chain, the class implements the strategy for generating new location
+    changes.
+    """
+    tail_limit = 1.e8
```

Comment:
I think constants should be named using all UPPER CASE by pep 8 convention.

Reply:
Adjusted.